docs/message-queues.asciidoc

```diff
@@ -0,0 +1,98 @@
+const sqs = new AWS.SQS({apiVersion: '2012-11-05'});
+
+/* ... */
+
+const transaction = apm.startTransaction("Process Messages", 'cli')
+sqs.receiveMessage(params, function(err, data) {
+  if(err) {
+    console.log("Receive Error", err);
+  } else {
+    console.log(`Length: ${data.Messages.length}`)
+    /* process messages */
+  }
+  // end the transaction
+  transaction.end()
+})
+----
+
+In the above code you can see we've started a transaction before calling the `sqs.receiveMessage` method, and ended it _after_ the queue's processing callback finishes executing.  The will ensure a transaction is active while processing your queue messages.
+
+[float]
+[[message-queues-distributed-tracing]]
+=== Distributed Tracing and Messaging Queues
+
+Similarly, if you want to connect your queue scheduling and queue processing via distributed tracing, you'll need to use the Node.js Agent's API to _store_ a `traceparent` header with your queue message and then provide that `traceparent` header when starting the new transaction.
+
+Here's an example that uses the Node.js Agent API to store the `traceparent` as a message attribute and then uses that attribute to link your new transaction with the original.
+
+First, when sending the message, we add the trace as one of the `MessageAttributes`
+[source,js]
+----
+// stores the traceparent when sending the queue message
+const traceParent = apm.currentTransaction ? apm.currentTransaction.traceparent : ''
+
+// Use the Amazon SQS `MessageAttributes` to pass 
+// on the  traceparent header
+const params = {
+  /* ... other params ... */
+  MessageAttributes: {
+    /* ... other attributes ... */
+    "MyTraceparent":{
+        DataType: "String",
+        StringValue: traceParent
+    }
+  }
+
+}
+sqs.sendMessage(params, function(err, data) {
+  /* ... */
+});
+----
+
+Then, when receiving the messages, we start and end our transaction while looping over the messages.  
+
+[source,js]
+----
+// uses the traceparent to start a transaction
+
+sqs.receiveMessage(params, function(err, data) {
+  if(!data.Messages) {
+    return
+  }
+  
+  // loop over your returned messages
+  for(const message of data.Messages) {
+    // start a transaction to process each message, using our previously
+    // saved distributed tracing traceparent header
+    let traceparent
+    if(message.MessageAttributes.MyTraceparent) {
+        traceparent = message.MessageAttributes.MyTraceparent.StringValue
+    }
+    const transactionMessage = apm.startTransaction('RECEIVE_TRANSACTION', 'cli', {
+      childOf:traceparent
+    })
+    /* ... process message ... */
+    transactionMessage.end()
+  }
+})
+
+----
+
+Unlike the first code sample, this approach will create a single transaction for each processed queue message.  This allows you to link each individual message with the transaction it started in. The <<apm-start-transaction,`startTransaction()`>> API allows _you_ to control how your queue data is reported to Elastic APM.
```

Comment:
I'm confused here too.

> Unlike the first code sample, ...

But the first example is about sending the message?

Reply:
@bmorelli25 OK, I see the confusion here.  It was not copy/pasta as I originally thought.  I wasn't referring to the immediately previous code sample, I was referring to the code sample at the _top_ of this page.  

I've pulled this last paragraph in favor of some callouts.  I also decided that trying to communicate the difference in where a transaction is started might impact how spans and linked might be too much information for this doc. 

If there's still confusion around this lets sync up via slack or zoom to talk it out.

Reply:
Doh! That makes sense. Let me take another read-through.